src/main/java/spark/route/RouteMatcher.java

```diff
@@ -0,0 +1,117 @@
+package spark.route;
+
+/**
+ * @author amarseillan
+ */
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import spark.utils.MimeParse;
+
+public abstract class RouteMatcher {
+	
+	private static final org.slf4j.Logger LOG = org.slf4j.LoggerFactory.getLogger(RouteMatcher.class);
+    private static final char SINGLE_QUOTE = '\'';
+
+	   /**
+     * Parse and validates a route and adds it
+     *
+     * @param route      the route path
+     * @param acceptType the accept type
+     * @param target     the invocation target
+     */
+    public void parseValidateAddRoute(String route, String acceptType, Object target) {
+    	try {
+            int singleQuoteIndex = route.indexOf(SINGLE_QUOTE);
+            String httpMethod = route.substring(0, singleQuoteIndex).trim().toLowerCase(); // NOSONAR
+            String url = route.substring(singleQuoteIndex + 1, route.length() - 1).trim(); // NOSONAR
+
+            // Use special enum stuff to get from value
+            HttpMethod method;
+            try {
+                method = HttpMethod.valueOf(httpMethod);
+            } catch (IllegalArgumentException e) {
+                LOG.error("The @Route value: "
```

Comment:
This PR is like a year old... Don't think it makes sense to make it better.

Things changed a lot, merging would be awful
:)